channel/fd_channel.cpp

```diff
@@ -0,0 +1,163 @@
+#include "fuzzuf/channel/fd_channel.hpp"
+#include "fuzzuf/logger/logger.hpp"
+#include "fuzzuf/utils/common.hpp"
+
+#include <signal.h>
+#include <stdarg.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/wait.h>
+#include <time.h>
+#include <unistd.h>
+
+FdChannel::FdChannel() {
+    // Nothing to do
+}
+
+FdChannel::~FdChannel() {
+    TerminateForkServer();
+}
+
+// Write exact `size` bytes
+ssize_t FdChannel::Send(void *buf, size_t size) {
+    ssize_t nbytes = Util::WriteFile(forksrv_write_fd, buf, size, true);
+    if (nbytes < 0) {
+        ERROR("[FdChannel] Failed to send");
```

Comment:
This should return -1 as it's used from `WaitForkServerStart`
```suggestion
        return -1;
```

Reply:
This suggestion was not applied. Final decision is https://github.com/fuzzuf/fuzzuf/pull/53/commits/b8685960ab7d47c21f71f2f2901dd77e9aedea1c .